iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/rules/MultiplexingClientTestRule.java

```diff
@@ -0,0 +1,50 @@
+package tests.integration.com.microsoft.azure.sdk.iot.helpers.rules;
+
+import com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationAuthenticationException;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assume;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.IntegrationTest;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.annotations.StandardTierHubOnlyTest;
+
+@Slf4j
+public class MultiplexingClientTestRule implements TestRule
+{
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new MultiplexingClientTestStatement(base, description);
+    }
+
+    private static class MultiplexingClientTestStatement extends Statement {
+
+        private final Statement base;
+
+        public MultiplexingClientTestStatement(Statement base, Description description) {
+            this.base = base;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
```

Comment:
This new rule provides a bit better error case logging when a multiplexing client test fails. Rather than having a nested exception, a MultiplexingClientDeviceRegistrationAuthenticationException has one child exception for each device that failed to authenticate. This new rule makes sure to print out those exceptions which will make debugging gate errors easier